src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -597,7 +632,10 @@ protected virtual string GenerateDisplayText(ModelMetadata metadata)
             return tagBuilder.ToHtmlString(TagRenderMode.Normal);
         }
 
-        protected virtual HtmlString GenerateEditor(ModelMetadata metadata, string htmlFieldName, string templateName,
+        protected virtual HtmlString GenerateEditor(
+            ModelMetadata metadata,
+            string htmlFieldName,
```

Comment:
ignore my previous comment.  `htmlFieldName` is an override, not necessarily (though usually) the same as the `expression` `string` the user passed or the name we generate for the lambda the user passed.